### ISSUE TITLE
"How we made Jupyter Docker Stacks multi-arch" Jupyter Medium Blog

### ISSUE THREAD

**Issue by mathbunnyru** [jupyterhub/team-compass#610]
### Proposed change
I would like to create a blog post on jupyter blog.
It will be about how we enabled support for aarch64 images.

### Alternative options
I think this is the best place for this story.

### Who would use this feature?
We will draw more attention to our Jupyter Docker Stacks.
This post will be for all the users of our Docker images if they want to use aarch64 images and didn't know they are finally supported (without any caveats or significant bugs).
This post will also be useful for all Docker image maintainers, who would like to make their images multi-arch and use GitHub as a CI system.

### (Optional): Suggest a solution
Here is a draft: https://medium.com/@mathbunnyru/how-we-made-jupyter-docker-stacks-multi-arch-part-1-accd9a481f6

**Comment by mathbunnyru** [comment]
@minrk @choldgraf @consideRatio could you please tell me, who should I ask to provide a light review and approve the article?
According to the [docs](https://jupyterhub-team-compass.readthedocs.io/en/latest/practices/external-communication.html#blog-posts), I should do it, right?

Also, I didn't have permission (and don't have one currently) to create drafts in Jupyter Blog, that's why I started it in my personal account. I hope it's possible to just move the story there.

**Comment by minrk** [comment]
@mathbunnyru I believe you should now have the option to 'add to publication' for the jupyter blog in the menu:

<img width="280" alt="Screenshot 2022-12-16 at 12 24 59" src="https://user-images.githubusercontent.com/151929/208088114-dfc9ea61-7736-41f6-b679-6fa4cf6bead5.png">

**Comment by mathbunnyru** [comment]
Thanks, I submitted to publication! 👍

**Comment by choldgraf** [comment]
Here's a quick thought: what do people think about us having a blog post review process like this to be a bit more clear:

- we keep a list of blog posts by the jupyterhub team in our team compass
- when you have a draft ready, open a PR adding to that list. That PR is where we request reviews from other team members. It can close the issue when merged if an issue exists.
- the PR description should have a link to the draft version of the post
- approving the PR is approving the post, and we follow the same guidelines as for code PRs.
- closing the PR then adds a link to the published post to our docs, closes the issue, and we're done

**Comment by mathbunnyru** [comment]
I think it's a reasonable approach to the review process.

I would also like to make some things a bit more clear [here](https://jupyterhub-team-compass.readthedocs.io/en/latest/practices/external-communication.html#blog-posts):

> When it’s ready, ask a team member to take a look

I wasn't sure who I should ask.

> Provide a light review, Approve the post via a comment, Post to the blog and close the issue

I think this is a bit confusing, because this process describes two processes: of the writer and of the editors and I think it would be best to have two sections instead of just one.

**Comment by choldgraf** [comment]
I took a quick look at the draft, and it looks good to me. I made a few small edits just for wording and clarity, nothing substantial. Here are a few general comments but they don't need to be blocking:

1. The post is quite long. In general I think it's good to shoot for like 500-600 words for a blog post, or you'll lose people's attention. This one clocks in at over 1500 words. I think it's OK if you want this to be a long-term technical reference or something, but just noting that this is a lot.
2. I suggest adding a short paragraph closer to the top with the solution you found in the end. I know that the end result isn't the goal of the post, but it takes a long time to get to the "punchline". I'd suggest asking yourself "what is the one or two most important takeaways that readers should have from this", and make sure that each of these is clearly stated within the first few paragraphs.
3. There seems to be some kind of weird whitespace / paragraph thing going on, but I'm not sure exactly what. There were a few places that I tried to consolidate paragraphs but it wouldn't let me add any spaces. Just noting it but maybe there's nothing actionable.

> I wasn't sure who I should ask.

That's because there's no official role for doing this :-) 

I think it's one reason that a PR in the team compass is helpful - at least that way we can have a clearer paper trail of "who is reviewing blog posts" and a signal to the team when somebody wants assistance.

> Provide a light review, Approve the post via a comment, Post to the blog and close the issue

Let's clean up the blogging language in a follow-up PR? It would be helpful for you to provide your perspective in cleaning up the language.

**Comment by mathbunnyru** [comment]
Thank you @choldgraf, that's awesome.

It would be nice to see all the changes you made in a git-diff like way.
And it would be even better to store the article as a markdown on GitHub and the review process would be just a normal PR review process on Github.
There are some tools to export markdown to medium, but they are not hadnling complicated markdown properly (as far as I understand) and I guess they have other issues as well.

**Comment by mathbunnyru** [comment]
@choldgraf

> 1. In general I think it's good to shoot for like 500-600 words for a blog post, or you'll lose people's attention.

Thank you. I will keep this in mind next time.

> 2. I suggest adding a short paragraph closer to the top with the solution you found in the end.

I added `If you want to read how it is implemented now, you can skip to the last section of this post.` to the top.
So, the people, who want to skip the history and the first try, can go straight to the implementation.

> 3. There seems to be some kind of weird whitespace / paragraph thing going on, but I'm not sure exactly what.

I tried to fix some places where it was possible.

Overall, I think this looks really good now and is ready to be posted 😄

**Comment by minrk** [comment]
Published. Thanks, @mathbunnyru!

**Comment by choldgraf** [comment]
Thanks @mathbunnyru for writing this post! And to @minrk for posting 🙂

**Comment by mathbunnyru** [comment]
Hi everyone.

I'm currently in the process of preparing an application for a UK Global Talent Visa and I will apply in a week.
I wanted to use this article as part of my application, but Tech Nation doesn't seem to like the Medium platform, so I would like to hide it for a month (I have a copy of this article on dev.to platform).
I plan to press "Remove story from publication", and then, as soon as I get an endorsement (or reject) from TechNation, I will return this story back.

Please, get back to me, if you don't think it's fine.
I want to make everyone happe and it seems to be the best decision I currently have.

**Comment by choldgraf** [comment]
For what it's worth I would totally support this

**Comment by manics** [comment]
Note I support it too (the UK immigration system is notoriously bad), but since it'll break a published post can we at least ensure all current blog members are aware of this?

**Comment by mathbunnyru** [comment]
Thanks everyone for a valuable feedback :heart:

> I don't quite understand why presence on the blog is a problem if it's also somewhere else, but your suggestion is totally fine with me if it helps you!

That's just their rules from here: https://technation.io/visa-tech-nation-visa-guide
_LinkedIn or Medium are not considered sufficient as evidence._

> Since this is a shared Jupyter blog (and not just JupyterHub) in the interests of transparency is there a more appropriate place we can have this discussion?

I think, as long as everyone is quite ok with my suggestion, let's keep it simple (and here).

> Note I support it too (the UK immigration system is notoriously bad), but since it'll break a published post can we at least ensure all current blog members are aware of this?

@manics could you please tell me, what should I do?
I thought I would just "remove story from publication", it would be hidden and then, in a month it will be back, so it shouldn't cause lots of trouble.
And to be honest, this post wasn't popular as other posts (I'm a bit sad, because I've spent quite some time writing it, but it happens), so I don't think I cause lots of damage.

**Comment by manics** [comment]
> Also note that evidence of media recognition should include details about the publication and target audience. LinkedIn or Medium are not considered sufficient as evidence. Internal company awards, training or certificates are also not sufficient for meeting this criterion. Where possible evidence should include supporting data such as number of page views, downloads or other quantifiable metrics.

Doesn't this just mean you can't use things like blog posts as evidence in your application? I don't think it's saying you're not allowed to post on LinkedIn/Medium... The immigration process has some strange rules but I don't think they'd go as far as saying you can never have posted on LinkedIn. I think by "media recognition" they're looking for industry publications, i.e. publishing work in a journal or website that's widely seen as a reliable source of news or information, or in a peer reviewed publication (not necessarily academic, just something with an editorial process), or (maybe?) being featured in a news article for the work you've done.

I won't block hiding the post if you still want to, but I don't think that's the intention of the guidance.

**Comment by mathbunnyru** [comment]
@manics you're totally right.
They are not saying "You can never have posted on LinkedIn or Medium".
But I have the same article on dev.to: https://dev.to/mathbunnyru/how-we-made-jupyter-docker-stacks-multi-arch-3ic1
And I was asked (by the agency, that is helping me to acquire the visa) to temporarily hide the one in the medium blog.

**Comment by choldgraf** [comment]
Yeah I think the point is that @mathbunnyru _wants_ to use this post as evidence of impact, but that the application guidelines specifically say that Medium posts are not going to count if that's what he points to.

**Comment by mathbunnyru** [comment]
> Yeah I think the point is that @mathbunnyru _wants_ to use this post as evidence of impact, but that the application guidelines specifically say that Medium posts are not going to count if that's what he points to.

Yep, I wanted to use my original blog post as evidence of impact, but then I was told Medium posts are not allowed (even though my original post was reviewed and edited by the Jupyter team), so I had to write it on another platform and I was asked to hide it from Medium while my application is reviewed.

**Comment by mathbunnyru** [comment]
Thanks everyone, I removed a story from publication. I hope it will disappear from google search soon.
Also, if someone can make this issue limited to collaborators, that would be great as well.
